clients/sail_ui/lib/rpcs/bitnames_rpc.dart

```diff
@@ -83,6 +83,96 @@ class BitnamesLive extends BitnamesRPC {
 
   @override
   Future<void> stopRPC() async {
-    await _client().call('stop');
+    try {
+      await _client().call('stop');
+      await Future.delayed(const Duration(milliseconds: 500));
+    } catch (e) {
+      if (e.toString().contains('Connection refused') || e.toString().contains('Unknown Error')) {
+        return;
+      }
+      log.e('Error stopping RPC', error: e);
+      rethrow;
+    }
+  }
+
+  @override
+  Future<void> setSeedFromMnemonic(String mnemonic) async {
+    try {
+      final home = Platform.environment['HOME'] ?? Platform.environment['USERPROFILE'];
+      if (home == null) {
+        throw Exception('Could not find home directory');
+      }
+
+      final bitnamesDir = path.join(home, 'Library', 'Application Support', 'plain_bitnames');
+      await _deleteWalletFiles(bitnamesDir);
+
+      int setSeedRetries = 0;
+      const maxSetSeedRetries = 5;
+      const setSeedRetryDelay = Duration(milliseconds: 500);
+      Exception? lastError;
+
+      while (setSeedRetries < maxSetSeedRetries) {
+        try {
+          await _client().call('set_seed_from_mnemonic', [mnemonic]);
+          break;
+        } catch (e) {
+          lastError = Exception(e.toString());
+          setSeedRetries++;
+          if (setSeedRetries == maxSetSeedRetries) {
+            throw lastError;
+          }
+          await Future.delayed(setSeedRetryDelay);
+        }
+      }
+
+      int retries = 0;
+      const maxRetries = 20;
+      const retryDelay = Duration(seconds: 2);
+
+      while (retries < maxRetries) {
+        try {
+          await _client().call('balance');
+          return;
+        } catch (e) {
+          if (e.toString().contains("wallet doesn't have a seed")) {
+            retries++;
+            await Future.delayed(retryDelay);
+            continue;
+          }
+          rethrow;
+        }
+      }
+      throw Exception('Failed to verify wallet was ready after $maxRetries attempts');
+    } catch (e) {
+      log.e('Error setting Bitnames seed', error: e);
+      rethrow;
+    }
+  }
+
+  Future<void> _deleteWalletFiles(String dataDir) async {
+    try {
+      final walletMdbDir = Directory(path.join(dataDir, 'wallet.mdb'));
+      if (await walletMdbDir.exists()) {
+        await walletMdbDir.delete(recursive: true);
+      }
+
+      final dataMdbDir = Directory(path.join(dataDir, 'data.mdb'));
+      if (await dataMdbDir.exists()) {
+        await dataMdbDir.delete(recursive: true);
+      }
+
+      final walletDat = File(path.join(dataDir, 'wallet.dat'));
+      if (await walletDat.exists()) {
+        await walletDat.delete();
+      }
+
+      final mnemonicTxt = File(path.join(dataDir, 'mnemonic.txt'));
+      if (await mnemonicTxt.exists()) {
+        await mnemonicTxt.delete();
+      }
+    } catch (e) {
+      log.e('Error deleting Bitnames wallet files', error: e);
+      rethrow;
+    }
```

Comment:
This is already implemented in the wipeWallet-function on each binary in binaries.dart. Could we use that instead?